src/Ledger/Ratify.lagda

```diff
@@ -231,77 +231,82 @@ restrictedDists coins rank dists = dists
         restrict : Credential ⇀ Coin → Credential ⇀ Coin
         restrict dist = topNDRepDist rank dist ∪ᵐˡ mostStakeDRepDist dist coins
 \end{code}
+\begin{code}[hide]
+module _ (Γ : RatifyEnv)(pparams : PParams) where
+
+  open RatifyEnv Γ
+  open PParams pparams
+\end{code}
+
 \begin{figure*}[h!]
+\begin{AgdaAlign}
 {\small
 \begin{code}
-actualVotes : RatifyEnv → CCData → (GovRole × Credential) ⇀ Vote → GovAction → PParams
-            → VDeleg ⇀ Vote
-actualVotes Γ cc votes ga pparams
-  =    mapKeys (credVoter CC) actualCCVotes
-  ∪ᵐˡ  actualPDRepVotes ∪ᵐˡ actualDRepVotes
-  ∪ᵐˡ  actualSPOVotes
-  where
-    open RatifyEnv Γ
-    open PParams pparams
+  activeDReps : ℙ Credential
+  activeDReps = dom $ filterᵐ? (λ x → currentEpoch ≤? (proj₂ x)) dreps
```

Comment:
Okay, but what you're describing @WhatisRT is exactly what I did already.  We had everything in a where clause and an item of issue #146 (PDF cleanup) requests a simplification of the proof and suggests removing some stuff from the where clause.  I 
simplified the proof and, to further simplify the presentation, I created the module and, indeed, I even added to the documentation a phrase like, "we assume a fixed ratify environment" (around line 299).

Reply:
The issue is that nothing in those changes justifies breaking up the `where` block. You did some simplifications, but you can just put all that new code back under the `where` block and it's fine. The issue I'm having with that figure isn't that it's a big **`where`** block, it's that it is a **big** `where` block. So what should be done about that? Well, definitions that don't necessarily need to be in that block can be moved out (`isHF` only depends on the governance action and can be general purpose, so that's a good candidate. Are there others?). Also, are there definitions that are used only once? If yes, does readability suffer if you inline it? If no, go ahead and inline it.